tests/shared/nlu/training_data/formats/test_rasa_yaml.py

```diff
@@ -519,3 +521,20 @@ def test_responses_text_multiline_is_preserved():
 
     # dumping again should also not change the format
     assert dumped == RasaYAMLWriter().dumps(dumped_result)
+
+
+def test_intent_examples_multiline_consistency():
+    """Test that multiline examples are written back as multiline examples."""
+
+    training_data_file = "data/test_multiline_intent_examples_yaml/nlu.yml"
+    training_data_from_disc = RasaYAMLReader().read(filename=training_data_file)
+
+    tmpfile = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
I'd suggest to use the `tmp_path` fixture from pytest. This why everything is cleaned up correctly after the test.

Reply:
nit picking: `pathlib` offers you a `Path.read_text(...)` which would make the code a tiny bit more concise